add barebones NPC's for chocobo circuit.

Tom-Neverwinter opened the pull request:
<!-- Remove space and place 'x' mark between square [] brackets or click the checkbox after saving to affirm the following points: -->
<!-- (it should look like this: - [x] I have ...) -->
**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)
- [x] I've _**tested my code and the things my code has changed**_ since the last commit in the PR, and will test after any later commits

## What does this pull request do?

<!-- Describe what your PR does here. If it closes an existing issue, you can mention: "Closes #1234" and GitHub will link this PR to that issue. -->

adds basic npcs to chocobo circuit as it only had a few that worked. This is not a functional set, this is to have the basic npcs spit out dialogue. zone is still busted, cant zone in, cant exit zone, some npcs are wrong, most essential cutscenes are busted

## Steps to test these changes

<!-- Clear and detailed steps to test your changes here -->

encounter npcs and activate.

Tom-Neverwinter committed:
Chocobo Circuit missing NPCs

This is a mess, needs style fixes. major functionality is missing.

However this does get many items limping along

zone needs to be fixed, some npcs are not correct.

Tom-Neverwinter committed:
Chocobo Circuit NPC's 

cleaned up the style a lot more, labeled the npcs, added a lot of the event id's

zach2good commented on scripts/zones/Chocobo_Circuit/npcs/Adrian.lua:
```diff
@@ -1,7 +1,8 @@
--- Standard Info NPC
+-- NPC: Adrian
+-- Abyssea Support NPC
+-- POS: -330.5837 0.0000 -468.5058
```
The rest of the codebase has the pos comment set so that you can copy/paste it to go straight to the NPC

Example:
```lua
-- !pos 230.413 32.278 280.677 15
```

Also (not that it matters much), they're all 3 decimal places or less

Tom-Neverwinter replied:
I added the zone value based on the !pos command `"func: pos <x> <y> <z> <optional zone> <optional target>` I will begin updating the others to match the three pos style

zach2good commented on scripts/zones/Chocobo_Circuit/npcs/Adrian.lua:
```diff
@@ -1,7 +1,8 @@
---  NPC: Adrian
--- Standard Info NPC
+-- NPC: Adrian
+-- Abyssea Support NPC
```
I assumed this was a copy/paste error, but no, there's Abyssea NPCs in here!

https://ffxiclopedia.fandom.com/wiki/Adrian

Tom-Neverwinter replied:
yup, same page I got the info from

zach2good commented on scripts/zones/Chocobo_Circuit/npcs/Alangriche.lua:
```diff
@@ -0,0 +1,22 @@
+-- Area: Chocobo_Circuit
+-- NPC: Alangriche
+-- Novice San d'Oria CRA Branch
+-- pos
```
All new NPCs will need their `!pos` comments filled

Tom-Neverwinter replied:
I don't know how to do an id shift?, this NPC is not loading into the map? Same for the Aht Urhgan Whitegate and [-151.2092 -0.0686 -388.0844] Windurst Woods [-140.2342 -0.0686 -534.2736] sections.

There is an area replaced by synthsizers which I also dont know how to correct [-313.1573 -0.0686 -519.3740]

zach2good commented on scripts/zones/Chocobo_Circuit/npcs/Alangriche.lua:
```diff
@@ -0,0 +1,22 @@
+entity.onEventFinish = function(player, csid, option)
+end
+
+return entity
```
CI is warning you of this: All files need to end with a newline

Tom-Neverwinter replied:
done :)

zach2good commented on scripts/zones/Chocobo_Circuit/npcs/Azainnie.lua:
```diff
@@ -0,0 +1,23 @@
+-- NPC: Azainnie
+-- Grandstand Exit
+-- pos -62.2576 -15.0000 -132.6927
+
```
Gap

Tom-Neverwinter replied:
Fixed :)

zach2good commented on scripts/zones/Chocobo_Circuit/npcs/Boirie.lua:
```diff
@@ -0,0 +1,23 @@
+Boirie
```
Typo

Tom-Neverwinter replied:
Fixed, will upload next

zach2good reviewed (changes requested):
CI will tell you most of the style and sanity changes you need to make

Do the CS's for all these NPCs make some sort of sense? Like, no empty menus, no CSs that need eventUpdates to function properly etc.

A kind-of-working-but-mostly-working zone is much worse than a zone that does nothing at all

Tom-Neverwinter committed:
Update Zone.lua

 --temp zone in testing --teleporters need scripting --zone in doors not scripted add globals

Tom-Neverwinter committed:
newline added

CI is warning you of this: All files need to end with a newline

Tom-Neverwinter committed:
missed a newline

Tom-Neverwinter committed:
remove trailing spaces notepad++ macro test

blanking operations

Tom-Neverwinter committed:
Merge branch 'LandSandBoat:base' into base

Tom-Neverwinter committed:
remove trailing spaces notepad++ macro test 2

Tom-Neverwinter committed:
Merge branch 'LandSandBoat:base' into base

Mortalelite commented on scripts/zones/Chocobo_Circuit/npcs/Maleroune.lua:
```diff
@@ -0,0 +1,24 @@
+Maleroune
```
This line needs fix. We would just remove the name as already listed

Tom-Neverwinter committed:
fix Maleroune

remove unneeded line

Tom-Neverwinter committed:
Update Zone.lua

Tom-Neverwinter committed:
Delete Odersille.lua

Tom-Neverwinter committed:
maybe this will fix it?

Tom-Neverwinter committed:
Update Zone.lua

Tom-Neverwinter committed:
last try.

I'm just wasting build cycles for other users productivity. I really dislike this ci being able to detect and unable to automatically fix the issue.

Mortalelite commented on scripts/zones/Chocobo_Circuit/npcs/Odersille.lua:
```diff
@@ -0,0 +1,22 @@
+entity.onEventFinish = function(player, csid, option)
+end
+
+return entity
```
As mention end file needs a new line

Sanity Check will let you know what needs to be fixed in most things

`  scripts/zones/Chocobo_Circuit/npcs/odersille.lua: No newline at end of file, please add one. `

Mortalelite commented on scripts/zones/Chocobo_Circuit/npcs/odersille.lua:
```diff
@@ -0,0 +1,22 @@
+entity.onEventFinish = function(player, csid, option)
+end
+
+return entity
```
As mention end file needs a new line

Sanity Check will let you know what needs to be fixed in most things

` scripts/zones/Chocobo_Circuit/npcs/odersille.lua: No newline at end of file, please add one. `

Tom-Neverwinter replied:
I know. git editor or notepad++ are a pain. its hard to work with invisible characters. [spending all day on something so trivial is not pleasant.]

Mortalelite replied:
I think you can also edit with Visual Studio. Someone can correct me if I am wrong on that part

TeoTwawki replied:
get the editorconfig plugin for notepad++ and a majority of its formatting problems get automated away.

Tom-Neverwinter committed:
Merge branch 'LandSandBoat:base' into base

Tom-Neverwinter committed:
Update zonelines.sql

Tom-Neverwinter committed:
Update Zone.lua

Tom-Neverwinter committed:
try git editor

Tom-Neverwinter committed:
Merge branch 'LandSandBoat:base' into base

Tom-Neverwinter committed:
Copy greeter moogle 

This is a copy of Port San dOria greeter moogle with csid changed.

Tom-Neverwinter committed:
update npc_list

Tom-Neverwinter committed:
Merge branch 'LandSandBoat:base' into base

Tom-Neverwinter committed:
Update Alangriche.lua

Tom-Neverwinter committed:
Update Chualulu.lua

Tom-Neverwinter committed:
Update Curtis.lua

Tom-Neverwinter committed:
Delete odersille.lua

Tom-Neverwinter committed:
exiting grandstands works!

where it exits to is not proper, they all exit to ilsorie for some reason? seems like the cutscene or event sets the exit point

Tom-Neverwinter committed:
Merge branch 'LandSandBoat:base' into base

Tom-Neverwinter committed:
update july 2022 + chocobo circuit

used npclogger_parse, cleaner and improved accuracy

Tom-Neverwinter committed:
Merge branch 'LandSandBoat:base' into base

Tom-Neverwinter committed:
copy pasta dealer moogle

Tom-Neverwinter committed:
copa pasta

Tom-Neverwinter committed:
Copy pasta

Tom-Neverwinter committed:
Copy pasta

Tom-Neverwinter committed:
Copy Pasta

Tom-Neverwinter committed:
add items from dealer_moodle branch
